File: app/src/main/java/de/hdc/kspchecklist/data/ListItem.java
package de.hdc.kspchecklist.data;

/**
 * Created by DerTroglodyt on 2017-02-13 12:20.
 * Email example@gmail.com
 * Copyright by HDC, Germany
 */

public class ListItem implements Comparable {

    public final String name;

    public static ListItem create(String name) {
        return new ListItem(name);
    }

    @Override
    public String toString() {
        return name;
    }

    private ListItem(String name) {
        this.name = name;
    }

    @Override
    public int compareTo(Object o) {
        return this.name.compareTo(o.toString());
    }
}

File: app/src/main/java/de/hdc/kspchecklist/data/ListItem.kt
package de.hdc.kspchecklist.data

/**
 * Created by DerTroglodyt on 2017-02-13 12:20.
 * Email example@gmail.com
 * Copyright by HDC, Germany
 */

class ListItem private constructor(val name: String) : Comparable<Any> {

    override fun toString(): String {
        return name
    }

    override operator fun compareTo(other: Any): Int {
        return this.name.compareTo(other.toString())
    }

    companion object {

        fun create(name: String): ListItem {
            return ListItem(name)
        }
    }
}
